src/Microsoft.AspNetCore.Routing/Matchers/IEndpointSelectorPolicy.cs

```diff
@@ -0,0 +1,12 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Routing.Matchers
+{
+    public interface IEndpointSelectorPolicy
+    {
+        void Apply(HttpContext httpContext, CandidateSet candidates);
```

Comment:
Do you think async policies are a thing people would want?

Reply:
Yeah, I was little unsure about that. We've traditionally not offered async at this level and it's been fine so far. You can always make an argument that things should be async, so I'm not really swayed by speculation.

Reply:
I've tagged this for discussion as part of API review.